File: Data Structures & Algorithms/sliding-window-maximum/submission-1.cpp
class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        multiset<int> cs;
        for (int i = 0; i < k-1; i++){
            cs.insert(nums[i]);
        }

        vector<int> res(nums.size() - k + 1);
        for (int i = k-1; i < nums.size(); i++){
            cs.insert(nums[i]);
            res[i-k+1] = *cs.rbegin();
            cs.erase(cs.find(nums[i-k+1]));
        }
        return res;
    }
};

File: Data Structures & Algorithms/valid-parenthesis-string/submission-1.py
class Solution:
    def checkValidString(self, s: str) -> bool:
        k = k2 = 0
        for c in s:
            if c == "(" or c == "*":
                k += 1
            else:
                k -= 1
            if k < 0:
                return False
        
        for c in s[::-1]:
            if c == "(":
                k2 += 1
            else:
                k2 -= 1
            if k2 > 0:
                return False

        return True

# (((((*(((((*((**(((*)*((((**))*)*)))))))))((*(((((**(**)